codec-http2/src/main/java/io/netty5/handler/codec/http2/HttpToHttp2ConnectionHandler.java

```diff
@@ -78,8 +78,11 @@ protected HttpToHttp2ConnectionHandler(Http2ConnectionDecoder decoder, Http2Conn
      * @throws Exception If the {@code httpHeaders} object specifies an invalid stream id
      */
     private int getStreamId(HttpHeaders httpHeaders) throws Exception {
-        return httpHeaders.getInt(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text(),
-                                  connection().local().incrementAndGetNextStreamId());
```

Comment:
wow... this looks like a "bug" before as we always incremented the stream id even tho there might be the header. Let me open a PR against 4.1

Reply:
actually never mind... this works as expected.